### PR TITLE
Grishniakova_send CV

### PR DESCRIPTION
**Pull request by KseniiaGrishn**

### CODE REVIEW

**Comment by KseniiaGrishn** [comment]
For review

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 1]
```diff
@@ -0,0 +1,82 @@
+Feature 1.1. Send CV directly from the careers main page careers.quantori.com
+
+Scenario User sent CV 
+Given User filled in valid alphabetic name and surname
+And User filled in valid e-mail
+And User attached one file .RTF less than 10 mb
+And User put a tick in the checkbox
+When User pressed Send
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:8]: It might be better to use the present simple for "when" and "then" to make it more logical, since this sequence of actions must happen every time with the "given" above.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+Then User's CV was sent and Thank you message displayed
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:9]: The submission of the user's CV and the appearance of the "Thank you" message seem to be two separate events. 
It could make sense to at least endure "Thank you" message separately through "And" after "Then". 
Or even maybe move the message into a different scenario.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+Scenario User did not fill in name
+Given User filled in valid surname with no name
+And User filled in valid e-mail
+And User attached one file .PDF less than 10 mb
+And User put a tick in the checkbox
+When User pressed Send
+Then the CV was not sent to Quantori and "Name" field higllighted with red with message "Invalid name format"
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:17]: I can understand what the scenario is about, but there is still a feeling that several results are kind of mixed in one. 
For example, the "Name" text field will be highlighted in red even before the user clicks the "Send" button, so maybe it could be helpfull to describe the behavior of the "Name" field separately as a singular, independent behavior (not mixed with "Send" button's behavior).

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+Scenario User filled in name and surname in wrong format
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:19]: It's a great idea to consider different options for invalid values ​​for the "Name" field. If the user enters only a name or data in numerical format, this leads to one result ("Invalid name format" and a red line). 
There is a possibility to combine these scenarios into one Scenario Outline with more Examples. It's shorter and also it could be easier to describe other invalid data options for one text field.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Then the CV was not sent to Quantori and "Name" field higllighted with red with message "Invalid name format"
+
+Scenario User filled in wrong email 
+Given User filled in valid alphabetic name and surname
+And User filled in invalid e-mail with no @
+And User attached one file .PDF less than 10 mb
+And User put a tick in the checkbox
+When User pressed Send
+Then the CV was not sent to Quantori and "E-mail" field higllighted with red with message "Invalid name format"
+
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:34]: The same about invalid data in the "Email" field - it could be better to combine these scenarios into Scenario Outline with examples and to separate its behavior from other fields and buttons since they don't effect given result (("Invalid name format" and a red line).

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+And User filled in invalid e-mail without dot in .com
+And User attached one file .DOC less than 10 mb
+And User put a tick in the checkbox
+When User pressed Send
+Then the CV was not sent to Quantori and "E-mail" field higllighted with red with message "Invalid name format"
+
+Scenario User attached large CV file 
+Given User filled in valid alphabetic name and surname
+And User filled in valid e-mail
+And User attached one file .PDF more than 10 mb
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:46]: It might be better to be a little bit more specific here and use boundary value analysis for a file size testing.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+And User put a tick in the checkbox
+When User pressed Send
+Then the CV was not sent to Quantori and message "Something went wrong" displayed
+
+Scenario User did not accept Personal data agreement
+Given User filled in valid alphabetic name and surname
+And User filled in valid e-mail
+And User attached one file .PDF more than 10 mb
+And User did not put a tick in the checkbox
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:55]: I'm sure the above meant that the user attached a file less than 10 MB - this needs to be rechecked and corrected.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+When User pressed Send
+Then User can not press "Send" button and "I agree to the personal data processing" checkbox higllighted with red
+
+Scenario User attached large CV file and put only name without surname
+Given User filled in invalid alphabetic name and surname
+And User filled in valid e-mail
+And User attached one file .PDF more than 10 mb
+And User put a tick in the checkbox
+When User pressed Send
+Then User can not press "Send" button and no message displayed
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:65]: Unfortunately, it's not clear for me from this scenario why user can't press "Send" button - because of the wrong CV file size or of the invalid data in the "Name" field. 
As we already know, when user attaches a file more then 10MB, message "Something went wrong" displays. And when data in the "Name" field is invalid, the field turns red and message "Invalid name format" appears. User can't click the "Send" button only when these two factors are combined?
Here again I can suggest to separate different CV form's elements behaviors.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+Scenario User attached large CV file and wrote a long message 
+Given User filled in valid alphabetic name and surname
+And User filled in valid e-mail
+And User attached one file .PDF more than 10 mb
+And User put a tick in the checkbox
+And User wrote a long message above 500 characters
+When User pressed Send
+Then the CV was not sent to Quantori and message "Something went wrong" displayed
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:74]: Did it happen because user wrote a message above 500 characters AND attached a file more than 10 MB or JUST because (s)he attached a file more than 10 MB?

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+Scenario User wrote a long message 
+Given User filled in valid alphabetic name and surname
+And User filled in valid e-mail
+And User attached one file .PDF less than 10 mb
+And User put a tick in the checkbox
```
Comment (jdfedotova) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:80]: I think there should be a line with user writing a too long message. 
But this is the end of the test, for sure the eyes are already tired of this place. :)

**Review by jdfedotova — CHANGES_REQUESTED** [review]
You did a good job! 
Those moments that, in my opinion, require changes are quite similar to each other. I would advise you to pay attention to the separation of the CV form elements functions and the possibility of using Scenario Outlines.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 1 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:8]: Thank you for higlighting this - I will take this in concideration!

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 2 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:9]: Yes, I realised this a bit late, but now I will learn from my mistake! Thanks!

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 3 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:17]: Yes, agree with you, logic is lacking here.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 4 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:19]: I was a bit unsure about how to use Outline in this case, but after reviewing one of our colleagues I know the wau of doing it.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 5 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:34]: Agreed!

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 6 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:46]: Do you mean to put exact value to the weight of the size? Ok, got it!

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 7 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:55]: Yes, that's correct! I forget to correct it after copy-pasting!

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 8 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:65]: Hmm, good question! But I think inability to press the button is related to Name field. I have to recheck it. Good point!

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 9 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:74]: As I wrote in above comment - agree that it should be separated.

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature (hunk 10 above)
Comment (KseniiaGrishn) [reply qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:80]: Oh ,yes , I meant it but missed it!

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 11]
```diff
@@ -0,0 +1,82 @@
+Feature 1.1. Send CV directly from the careers main page careers.quantori.com
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:1]: Feature name shouldn't be long (if not necessary). Could name it "Send CV"

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+Scenario User sent CV 
+Given User filled in valid alphabetic name and surname
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:4]: ":" should be written after "Scenario"

Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:4]: Where is user and where he had filled name and surname?

Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:4]: Should be separated into two steps

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+And User filled in valid e-mail
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:5]: What is valid email?

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+And User attached one file .RTF less than 10 mb
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:6]: where from?

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+And User put a tick in the checkbox
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:7]: better say "user checked/selected "..." checkbox"

Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:7]: Which checkbox?

Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:7]: should be in Past tense

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+When User pressed Send
+Then the CV was not sent to Quantori and "Name" field higllighted with red with message "Invalid name format"
+
+Scenario User filled in name and surname in wrong format
+Given User filled invalid numeric name and surname
+And User filled in valid e-mail
+And User attached one file .PDF less than 10 mb
+And User put a tick in the checkbox
+When User pressed Send
+Then the CV was not sent to Quantori and "Name" field higllighted with red with message "Invalid name format"
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:25]: How we can check it?

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+When User pressed Send
+Then the CV was not sent to Quantori and message "Something went wrong" displayed
+
+Scenario User wrote a long message 
+Given User filled in valid alphabetic name and surname
+And User filled in valid e-mail
+And User attached one file .PDF less than 10 mb
+And User put a tick in the checkbox
+When User pressed Send
+Then the CV was not sent to Quantori and message "Something went wrong...Check the message field" displayed
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:82]: Missed case where user haven't filled anything

File: qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+Given User filled in valid alphabetic name and surname
+And User filled in invalid e-mail with no @
+And User attached one file .PDF less than 10 mb
+And User put a tick in the checkbox
+When User pressed Send
+Then the CV was not sent to Quantori and "E-mail" field higllighted with red with message "Invalid name format"
+
+Scenario User filled in email address with wrong format
+Given User filled in valid alphabetic name and surname
+And User filled in invalid e-mail without dot in .com
```
Comment (Georgiy-Petrov-Quantori) [inline qa/automation/CareerPortal.Tests.Scenarios/1. Access to Career Portal/Grishniakova Send CV test/Feature 1.1. Send CV direct from careers.feature:37]: What is invalid email?

**Review by Georgiy-Petrov-Quantori — CHANGES_REQUESTED** [review]
